bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: Conditionally set OAuth TLS certificate attribute on ManagedKafka type

## Description

This PR allows not sending a MAS SSO custom certificate on the ManagedKafka type.
The behavior is based on the `--mas-sso-cert-file` value and it is the following one:
* If `--mas-sso-cert-file` is not explicitely provided it will default to `secrets/keycloak-service.crt`. In that case:
  * If secrets/keycloak-service.crt exists then it will be used as the MAS SSO custom certificate
  * If secrets/keycloak-service.crt does not exist then it will continue without erroring and it will not use any MAS SSO custom certificate. It will also log this occurrence
* If `--mas-sso-cert-file` is explicitely provided:
  * If the provided file name exists and it is not empty then it will be used as the MAS SSO custom certificate
  * If the provided file name exists and it is empty then it will not use any MAS SSO custom certificate
  * If the provided file name does not exist then it will continue without erroring and it will not use any MAS SSO custom certificate. It will also log this occurrence

This uncommon behavior of "not using a custom mas sso certificate when file cannot be found" instead of erroring has been intentional. The reason for that is that that flag is always used in our service-template independently of the environment where it is deployed and OCP templates do not allow conditional statements.

## Verification Steps

1. an OCM cluster is needed
1. add it into your local kas fleet manager db
1. edit the secret `secrets/keycloak-service.crt` and put there a dummy value
1. start local kas fleet manager with the -v=10 flag and create a kafka request assigned it to the ClusterID that has been added to the DB. Then get the ClusterID of the cluster where tha kafka_request has been placed. To get the ID you can take a look at the kafka_request database entry
1.  manually call the /api/managed-services-api/v1/agent_clusters/<id>/kafkas endpoint. Remember that to call this endpoint the OCM token to provide has to be the one corresponding to the agent-cluster and not your user's token
1. verify that in the results there are the .spec.oauth.tls_trusted_certificate and .spec.oauth.tlsTrustedCertificate attributes and that both contain the previously set corresponding dummy values. 

Repeat all the previous steps but before starting kas fleet manager delete the secrets/keycloak-service.crt file. In this case you should verify that the  .spec.oauth.tls_trusted_certificate and .spec.oauth.tlsTrustedCertificate are NOT there anymore as part of the results. In kas fleet manager logs you should also see a log line with a log telling that the file could not be found.

Repeat all the previous steps but now when starting kas fleet manager start it with the --mas-sso-cert-file with a dummy value  to a non existing file. In this case you should verify that the  .spec.oauth.tls_trusted_certificate and .spec.oauth.tlsTrustedCertificate are NOT there anymore as part of the results. In kas fleet manager logs you should also see a log line with a log telling that the file could not be found.

Repeat all the previous steps but now have the secrets/keycloak-service.crt file existing but no content. In this case you should verify that the  .spec.oauth.tls_trusted_certificate and .spec.oauth.tlsTrustedCertificate are NOT there anymore as part of the results.

## Checklist (Definition of Done)

~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has created for changes required on the client side~~
